build.gradle.kts

```diff
@@ -24,27 +29,96 @@ kotlin {
         testRuns["test"].executionTask.configure {
             useJUnitPlatform()
         }
-    }
-    val iosTarget: (String, KotlinNativeTarget.() -> Unit) -> KotlinNativeTarget =
-        if (System.getenv("SDK_NAME")?.startsWith("iphoneos") == true)
-            ::iosArm64
-        else
-            ::iosX64
-
-    iosTarget("ios") {
-        binaries {
-            framework {
-                baseName = "JsonLogicKMP"
-                isStatic = true
+
+        val iosTarget: (String, KotlinNativeTarget.() -> Unit) -> KotlinNativeTarget =
+            if (System.getenv("SDK_NAME")?.startsWith("iphoneos") == true)
+                ::iosArm64
+            else
+                ::iosX64
+
+        iosTarget("ios") {
+            binaries {
+                framework {
+                    baseName = "JsonLogicKMP"
+                    isStatic = true
+                }
             }
         }
     }
     sourceSets {
-        val commonMain by getting
-        val commonTest by getting
-        val jvmMain by getting
-        val jvmTest by getting
-        val iosMain by getting
-        val iosTest by getting
+        val commonTest by getting {
+            dependencies {
+                implementation(kotlin("test"))
+            }
+        }
+    }
+}
+
+detekt {
+    buildUponDefaultConfig = true
+    autoCorrect = true
+    source = files("src/")
+    ignoreFailures = false
+    config = files("$projectDir/config/detekt/detekt.yml")
+}
+
+tasks.withType<Detekt>().configureEach {
+    reports {
+        html.required.set(true)
+        xml.required.set(true)
+        txt.required.set(true)
+        sarif.required.set(true)
+    }
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            from(components["java"])
+
+            pom {
+                name.set("JsonLogicKMP")
+                description.set("Kotlin multiplatform JsonLogic expressions evaluation engine")
+                url.set("https://github.com/allegro/json-logic-kmp")
+                inceptionYear.set("2022")
+                licenses {
+                    license {
+                        name.set("The Apache License, Version 2.0")
+                        url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+                    }
+                }
+                developers {
```

Comment:
should you be signed here specifically? Or we need more people/ more global option?